2-completed-app/AzureTablesDemoApplication/src/main/java/com/data/tables/service/impl/TablesServiceImpl.java

```diff
@@ -0,0 +1,144 @@
+package com.data.tables.service.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import com.azure.data.tables.models.TableTransactionAction;
+import com.azure.data.tables.models.TableTransactionActionType;
+import com.data.tables.data.SampleWeatherData;
+import com.data.tables.entities.ExpandableWeatherObject;
+import com.data.tables.entities.UpdateWeatherObject;
+import com.data.tables.models.FilterResultsInputModel;
+import com.data.tables.models.SampleDataInputModel;
+import com.data.tables.models.WeatherDataModel;
+import com.data.tables.models.WeatherInputModel;
+import com.data.tables.service.ITablesService;
+import com.data.tables.untils.WeatherDataUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.util.StringUtils;
+
+
+@Service
+public class TablesServiceImpl implements ITablesService {
+
+    @Autowired
+    private TableClient tableClient;
+    /**
+     * Get all weather data entities
+     * @return List<WeatherDataModel>
+     */
+    @Override
+    public List<WeatherDataModel> getAllRows() {
+        List<WeatherDataModel> modelList = new ArrayList<>();
+        tableClient.listEntities().stream().forEach(entity -> modelList.add(WeatherDataUtils.mapTableEntityToWeatherDataModel(entity)));
+        return Collections.unmodifiableList(WeatherDataUtils.filledValue(modelList));
```

Comment:
nit; We can make this look a bit cleaner by doing the following:
```suggestion
        List<WeatherDataModel> modelList = tableClient.listEntities().stream()
            .map(WeatherDataUtils::mapTableEntityToWeatherDataModel)
            .collect(Collectors.toList());
        return Collections.unmodifiableList(WeatherDataUtils.filledValue(modelList));
```